server-spi-private/src/main/java/org/keycloak/events/admin/AdminEvent.java

```diff
@@ -17,11 +17,26 @@
 
 package org.keycloak.events.admin;
 
+import org.keycloak.storage.SearchableModelField;
+
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
 public class AdminEvent {
 
+    public static class SearchableFields {
+        public static final SearchableModelField<AdminEvent> ID              = new SearchableModelField<>("id", String.class);
+        public static final SearchableModelField<AdminEvent> REALM_ID        = new SearchableModelField<>("realmId", String.class);
+        public static final SearchableModelField<AdminEvent> TIME            = new SearchableModelField<>("time", String.class);
+        public static final SearchableModelField<AdminEvent> AUTH_REALM_ID   = new SearchableModelField<>("authRealmId", String.class);
+        public static final SearchableModelField<AdminEvent> AUTH_CLIENT_ID  = new SearchableModelField<>("authClientId", String.class);
+        public static final SearchableModelField<AdminEvent> AUTH_USER_ID    = new SearchableModelField<>("authUserId", String.class);
+        public static final SearchableModelField<AdminEvent> AUTH_IP_ADDRESS = new SearchableModelField<>("authIpAddress", String.class);
+        public static final SearchableModelField<AdminEvent> OPERATION_TYPE  = new SearchableModelField<>("operationType", OperationType.class);
+        public static final SearchableModelField<AdminEvent> RESOURCE_TYPE   = new SearchableModelField<>("resourceType", ResourceType.class);
+        public static final SearchableModelField<AdminEvent> RESOURCE_PATH   = new SearchableModelField<>("resourcePath", String.class);
```

Comment:
Please add `EXPIRATION` similar to e.g. auth sessions
https://github.com/keycloak/keycloak/blob/5d87cdf1c6d420ece4691495701f956773fc2c43/server-spi/src/main/java/org/keycloak/sessions/RootAuthenticationSessionModel.java#L37

Reply:
Admin events currently don't have any expiration. It is possible to configure expiration only for auth events.